Source/Urho3D/Engine/Engine.cpp

```diff
@@ -63,8 +63,25 @@
 #include "../Urho2D/Urho2D.h"
 #endif
 
-#if defined(__EMSCRIPTEN__) && defined(URHO3D_TESTING)
+#ifdef __EMSCRIPTEN__
+#include "../Graphics/GraphicsEvents.h"
 #include <emscripten/emscripten.h>
+#include <emscripten/bind.h>
+static const Urho3D::Context *appContext;
+
+static void JSCanvasSize(int width, int height, bool fullscreen, float scale)
+{
+    using namespace Urho3D;

```

Comment:
Let's add error checks for null `appContext` and `ui`, just in case.
This function _technically_ can be called at any point of app lifetime.
And log `fullscreen` and `scale`, too.

Reply:
will do